src/foam/nanos/ruler/RulerDAO.js

```diff
@@ -140,7 +140,6 @@ if ( oldObj == null ) {
 // If the object was modified in an 'after' rule, then it is 'put'.
 FObject ret =  getDelegate().put_(x, obj);
 if ( ret != null ) {
-  ret = ret.fclone();
```

Comment:
Why was this here? Is it safe to remove? The applyRules can try to modify the object, but before the fclone() ret is frozen and can't be modified.

Reply:
Rules are not responsible for 'putting' any changes. Also, if a rule does 'put', then any subsequent rules will receive the pre-put object.  
So rules should receive an un-frozen object which they can modify.